src/components/main-view/main-view.jsx

```diff
@@ -140,8 +94,33 @@ export const MainView = () => {
         // });
 
         // userFavIDs = JSON.stringify(userData.Favorites);
-        setUserFavIDs(JSON.stringify(userData.Favorites));
+        setUserFavIDs(userData.Favorites);
+
+    }
 
+    async function logFav(id) {
+      /* 
+      Example of way to do this in memory.
+      Can you think of a way to optimize this code to run in batches using localStorage as an intermediary data structure?
+
+      const updatedUserFavIds = userFavIDs.includes(id) 
+        ? userFavIDs.filter((currId) => currId !== id) 
+        : [...userFavIDs, id];
+      setUserFavIDs(updatedUserFavIds); 
+      */
+
+      const isFavorite = userFavIDs.includes(id);
+      const method = isFavorite ? "DELETE" : "POST";
+
+      const response = await fetch("https://hidden-sea-19542.herokuapp.com/users/" + user.Username + "/movies/" + id, {
+          method,
+          headers: {
+              "Content-Type": "application/json",
+              "Authorization": "Bearer " + token
+          }
+      })
+
+      datafromAPI();
```

Comment:
This could be improved by returning the updated favorites array from in the API response instead of having to re-calculate it via calling this method